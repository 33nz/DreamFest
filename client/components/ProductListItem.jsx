import React from 'react'

const ProductListItem = (props) => {
  const product = props.product
  return (
    <div className='product'>
      <p className='name'>{product.name}</p>
      <p className='description'>{product.description}</p>
      <p>
        <span className='country'>Lovingly made in {product.country}</span>
        <button
          className='cart-link'
          onClick={() => props.addToCart(product)}>Add to cart</button>
      </p>
    </div>
  )
}

export default ProductListItem
