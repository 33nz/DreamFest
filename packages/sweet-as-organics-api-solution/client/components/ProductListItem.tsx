<<<<<<< HEAD:packages/sweet-as-organics-api-solution/client/components/ProductListItem.tsx
import { Product } from '../../common/interfaces'

interface Props {
  addToCart: (product: Product) => void
  product: Product
}
=======
import type { Product } from '../../common/Product'

interface Props {
  product: Product
  addToCart: (product: { id: number; name: string }) => void
}

function ProductListItem(props: Props) {
  const product = props.product

  function addToCart() {
    props.addToCart(product)
  }
>>>>>>> main:packages/sweet-as-organics-api/client/components/ProductListItem.tsx

function ProductListItem({ addToCart, product }: Props) {
  return (
    <div className="product">
      <p className="name">{product.name}</p>
      <p className="description">{product.description}</p>
      <p>
        <span className="country">Lovingly made in {product.country}</span>
        <button className="cart-link" onClick={() => addToCart(product)}>
          Add to cart
        </button>
      </p>
    </div>
  )
}

export default ProductListItem
