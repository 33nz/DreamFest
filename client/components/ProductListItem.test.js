import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import ProductListItem from './ProductListItem'

const product = {
  id: 1,
  name: 'mocked product',
  description: 'snazzy product',
  country: 'testlandia'
}

test('renders product correctly', () => {
  const { asFragment } = render(<ProductListItem product={product} />)
  expect(asFragment()).toMatchSnapshot()
})

test('calls addToCart on add to cart button click', async () => {
  const addToCart = jest.fn()
  render(<ProductListItem product={product} addToCart={addToCart} />)

  const addButton = await screen.getByRole('button', { name: 'Add to cart' })
  fireEvent.click(addButton)

  expect(addToCart).toHaveBeenCalled()
})
