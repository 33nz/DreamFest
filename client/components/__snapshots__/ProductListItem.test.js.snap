// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders product correctly 1`] = `
<DocumentFragment>
  <div
    class="product"
  >
    <p
      class="name"
    >
      mocked product
    </p>
    <p
      class="description"
    >
      snazzy product
    </p>
    <p>
      <span
        class="country"
      >
        Lovingly made in testlandia
      </span>
      <button
        class="cart-link"
      >
        Add to cart
      </button>
    </p>
  </div>
</DocumentFragment>
`;
